x-pack/packages/ml/inference_integration_flyout/components/connect_to_api.tsx

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { EuiSuperSelect } from '@elastic/eui';
+
+import React, { useMemo, useState } from 'react';
+import { connectToApiOptions, isFieldEmpty, setModalConfigResponse } from '../lib/shared_values';
+import type { ModelConfig } from '../types';
+import { Service } from '../types';
+import { InferenceFlyout } from './flyout_layout';
+import type { SaveMappingOnClick } from './inference_flyout_wrapper';
+import { CohereForm } from './service_forms/cohere_form';
+import { HuggingFaceForm } from './service_forms/huggingface_form';
+import { OpenaiForm } from './service_forms/openai_form';
+
+interface Props extends SaveMappingOnClick {
+  description: string;
+}
+export const ConnectToApi: React.FC<Props> = ({ description, onSaveInferenceEndpoint }) => {
+  const defaultOpenaiUrl = 'https://api.openai.com/v1/embeddings';
+  const defaultCohereModelId = 'embed-english-v2.0';
+
+  const [selectedModelType, setSelectedModelType] = useState(connectToApiOptions[0].value);
+
+  const [huggingFaceApiKey, setHuggingFaceApiKey] = useState('');
+  const [huggingFaceModelUrl, setHuggingFaceModelUrl] = useState('');
+
+  const [cohereApiKey, setCohereApiKey] = useState('');
+  const [cohereModelId, setCohereModelId] = useState(defaultCohereModelId);
+
+  const [openaiApiKey, setOpenaiApiKey] = useState('');
+  const [openaiEndpointlUrl, setOpenaiEndpointUrl] = useState(defaultOpenaiUrl);
+  const [openaiOrganizationId, openaiSetOrganizationId] = useState('');
+  const [openaiModelId, setOpenaiModelId] = useState('');
+
+  // disable save button if required fields are empty
+  const isSaveButtonEmpty = useMemo(() => {
+    if (selectedModelType === Service.huggingFace) {
+      return isFieldEmpty(huggingFaceModelUrl) || isFieldEmpty(huggingFaceApiKey);
+    } else if (selectedModelType === Service.cohere) {
+      return isFieldEmpty(cohereApiKey);
+    } else {
+      // open ai
+      return isFieldEmpty(openaiApiKey) || isFieldEmpty(openaiModelId);
+    }
```

Comment:
Added a form validation to prevent saving inference id, when required fields are empty.